### DISCUSSION TITLE
[Question] Soft Robot Simulation

### DISCUSSION THREAD

**Discussion by em-ni** [isaac-sim/IsaacLab#1327] [Q&A]
Hi,

Are there ways to actuate a deformable object, or they can be only passive?
For example, can I regulate the pressure of a fluid inside a soft object to control its deformation?

Thanks

**Comment by glvov-bdai** [comment]
Hi,

To the best of my knowledge, all deformable objects are passive. I've requested that NVIDIA add support for attaching deformable objects to rigid articulations, which might be able to control deformation in a future release by applying variable pressure, but I think this is still in development or backlog of the PhysX team.

**Reply by em-ni** [reply]
Thank you! 
is it possible to contribute to it?

**Comment by StrainFlow** [comment]
You can attach rigid bodies to deformable bodies in Omniverse, here's the documentation page that includes that: 

https://docs.omniverse.nvidia.com/extensions/latest/ext_physics/deformable-bodies.html

I have seen people create a rigid joint and then attach a deformable body to the rigid bodies on both sides of the joint, and then use that joint to make the deformable body move, but I'm not sure if this would work in Isaac Lab. I'm pinging some people about that now to see if it's possible.

**Comment by StrainFlow** [comment]
I've heard back, anything that works in USD and OV should work in Isaac Lab, so you can either model this soft body simulation in high fidelity using WARP or you could approximate the joint by attaching a deformable body to a rigid articulation. Here's a nice tutorial on how to spawn deformable bodies in Isaac Lab: 

https://isaac-sim.github.io/IsaacLab/main/source/tutorials/01_assets/run_deformable_object.html

**Comment by glvov-bdai** [comment]
@StrainFlow I don't think that attaching a deformable object to a rigid articulation would currently work in Isaac lab; have you tried this? If so, can you please provide the snippet (I am interested in this functionality).

In my experience, USD <-> Isaac Lab support is not 1:1.

From 

source/extensions/omni.isaac.lab/assets/articulations/articulation.py

```
An articulation is comprised of multiple rigid bodies or links. For a rigid body, there are two frames
    of reference that are used:
```

Articulations seem to use the physx.ArticulationView, but I think the current articulation implementation does not support deformable objects.

**Reply by StrainFlow** [reply]
That's entirely possible, I've seen it done with USD joints in Omniverse but haven't seen it done with an articulation.

**Reply by StrainFlow** [reply]
I'll try to put together a sample to confirm or deny this

**Reply by garylvov** [reply]
Thank you!

**Reply by saulgooodman** [reply]
> I'll try to put together a sample to confirm or deny this

Hello, has it been confirmed that soft bodies can be attached to articulation objects in Isaac-Sim? Thankyou

**Reply by SantiDiazC** [reply]
I am interested in this functionality, for a simplified model of a continuum robot. Also as mentioned before a more realistic model would be to use warp, but I do not know how to integrate it on isaaclab, especially when attached to rigid bodies or other articulations, for example a robot manipulator. would be great if there is any update on this for either the first case (attaching a deformable body to a rigid joint), or the second (create the model with warp and integrate it to isaaclab). Thank you in advance

**Reply by garylvov** [reply]
@StrainFlow

**Reply by StrainFlow** [reply]
I wasn't able to get attachments to soft bodies working when I looked into it before, but it looks like there's been an update to them. I'm looking into that now :)

**Reply by SantiDiazC** [reply]
Any progress on this Issue?
